From: Michael Cree <mcree@orcon.net.nz>
Date: Thu, 10 Sep 2015 20:40:54 -0300
Subject: Fix test-suite failure on Alpha

Pulseaudio fails to build on the Alpha architecture due to a failure
in the volume-test of the test suite. The failure in volume-test occurs
because it is compiled with -ffast-math which implies
-ffinite-math-only of which the gcc manual states that it optimizes
for floating-point arithmetic with the assumption that arguments and
results are not NaNs or +/-infinity, and futher notes that it may
result in incorrect output.

On the Alpha platform that is somewhat an understatement as the use of
non-finite floating-point arithmetic with -ffinite-math-only results in
a floating-point exception and the termination of the program.

The volume-test converts volumes into decibels (so a zero volume
becomes a negative infinity) and proceeds to add two volumes (in
decibels), thus does arithmetic with non-finite floating point numbers
despite being compiled with -ffast-math!
---
 src/tests/volume-test.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/tests/volume-test.c b/src/tests/volume-test.c
index bd0b01c..76b8206 100644
--- a/src/tests/volume-test.c
+++ b/src/tests/volume-test.c
@@ -114,7 +114,10 @@ START_TEST (volume_test) {
             double q, qq;
 
             p = pa_sw_volume_multiply(v, w);
-            qq = db + db2;
+	    if (isfinite(db) && isfinite(db2))
+		qq = db + db2;
+	    else
+		qq = -INFINITY;
             p2 = pa_sw_volume_from_dB(qq);
             q = l*t;
             p1 = pa_sw_volume_from_linear(q);
